cnxpublishing/db.py

```diff
@@ -1232,6 +1232,42 @@ def remove_acl(cursor, uuid_, permissions):
                        (uuid_, uid, permission,))
 
 
+def _upsert_persons(cursor, person_ids, lookup_func):
+    """Upsert's user info into the database.
+    The model contains the user info as part of the role values.
+    """
+    person_ids = list(set(person_ids))  # cleanse data
+
+    # Check for existing records to update.
+    cursor.execute("SELECT personid from persons where personid = ANY (%s)",
+                   (person_ids,))
+    try:
+        existing_person_ids = [x[0] for x in cursor.fetchall()]
+    except TypeError:
```

Comment:
Coveralls failed because this exception is not being used in our unit tests.  I can't think of a quick way to write a test for it.  Should we set the coveralls configuration file to ignore this exception?